server/get_teams_data.go

```diff
@@ -0,0 +1,281 @@
+package main
+
+import (
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/metrics"
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/msteams"
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/msteams/clientmodels"
+
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/store/storemodels"
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+const (
+	ErrorNoRowsInResult = "sql: no rows in result set"
+)
+
+func (p *Plugin) GetMSTeamsTeamAndChannelDetailsFromChannelLinks(channelLinks []*storemodels.ChannelLink, userID string, checkChannelPermissions bool) (msTeamsTeamIDsVsNames map[string]string, msTeamsChannelIDsVsNames map[string]string, errorsFound bool) {
+	msTeamsTeamIDsVsChannelsQuery := make(map[string]string)
+	msTeamsTeamIDsVsNames = make(map[string]string)
+	msTeamsChannelIDsVsNames = make(map[string]string)
+
+	for _, link := range channelLinks {
+		if checkChannelPermissions && !p.API.HasPermissionToChannel(userID, link.MattermostChannelID, model.PermissionCreatePost) {
+			p.API.LogInfo("User does not have the permissions for the requested channel", "UserID", userID, "ChannelID", link.MattermostChannelID)
+			continue
+		}
+
+		msTeamsTeamIDsVsNames[link.MSTeamsTeamID] = ""
+		msTeamsChannelIDsVsNames[link.MSTeamsChannelID] = ""
+
+		// Build the channels query for each team
+		if msTeamsTeamIDsVsChannelsQuery[link.MSTeamsTeamID] == "" {
+			msTeamsTeamIDsVsChannelsQuery[link.MSTeamsTeamID] = "id in ("
```

Comment:
I see this parenthesis is closed in line 78, but doesn't it make more sense to close it in this function (around line 44 I guess)? Why is it that we close it so far away?

Reply:
We need to run an extra loop on the map to add the closing bracket, which can be avoided by closing it in the [line](https://github.com/mattermost/mattermost-plugin-msteams-sync/blob/620e6723c8215ccb864580e26cb47a1e95ddb597/server/get_teams_data.go#L77).